semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,456 @@
+    for data in data_parameters:
+        data_all.addNode(data)
+    # Add all obs to be used in this updating scheme
+    obsdata = ert.getLocalConfig().createObsdata("OBS")
+    for obs in obs_group:
+        obsdata.addNode(obs)
+    # Attach the created dataset and obsset to the ministep
+    ministep.attachDataset(data_all)
+    ministep.attachObsset(obsdata)
+    # Then attach the ministep to the update step
+    facade.get_update_step().attachMinistep(ministep)
+
+    # Perform update analysis
+    ert.analysisConfig().set_log_path(output_path)
+    run_context = ErtRunContext.ensemble_smoother_update(
+        ert.getEnkfFsManager().getFileSystem(prior_name),
+        ert.getEnkfFsManager().getFileSystem(target_name),
+    )
+    ESUpdate(ert).smootherUpdate(run_context)
+
+
+def _get_field_params(ert, ensemble_size, field_parameters, target_name):
+    """
+    Because the FIELD parameters are not exposed in the Python API we have to
+    export them to file and read them back again. When they are exposed in the API
+    this function should be updated.
+    """
+    field_data = {}
+    file_system = ert.getEnkfFsManager().getFileSystem(target_name)
+    with tempfile.TemporaryDirectory() as fout:
+        for field_param in field_parameters:
+            config_node = ert.ensembleConfig()[field_param]
+            ext = config_node.get_enkf_outfile().rsplit(".")[-1]
+            file_path = os.path.join(fout, "%d_" + field_param + "." + ext)
+            _export_field_param(config_node, file_system, ensemble_size, file_path)
+            fnames = glob.glob(os.path.join(fout, "*" + field_param + "*"))
+            field_data[field_param] = _import_field_param(
+                ert.eclConfig().get_gridfile(), field_param, fnames
+            )
+    return field_data
+
+
+def _import_field_param(input_grid, param_name, files):
+    mygrid_char = Grid(input_grid.rsplit(".", 1)[0], fformat="eclipserun")
+    all_input = []
+    for file_path in files:
+        proproff = GridProperty(file_path, name=param_name, grid=mygrid_char)
+        array_nb = proproff.get_npvalues1d(activeonly=False, fill_value=0, order="C")
+        all_input.append(array_nb)
+    return all_input
+
+
+def _export_field_param(config_node, file_system, ensemble_size, output_path):
+    # Get/export the updated Field parameters
+    EnkfNode.exportMany(
+        config_node,
+        output_path,
+        file_system,
+        np.arange(0, ensemble_size),
+    )
+
+
+def initialize_emptydf():
+    ks_data = pd.DataFrame()
+    active_obs = pd.DataFrame()
+    misfitval = pd.DataFrame()
+    return ks_data, active_obs, misfitval
+
+
+def make_update_log_df(update_log_file, key):
+    """ Read update_log file to get active and inactive observations """
+    list_of_files = [
+        os.path.join(update_log_file, f) for f in os.listdir(update_log_file)
+    ]
+    if len(list_of_files) > 1:
+        raise OSError("ERROR more than one update_log_file.")
```

Comment:
As we are making the paths to updatelog from tmpdirs, this would really be a bug, right? I don't mind having it in though, just checking.

Reply:
Yes it would, though it is more of a remnant of the old implementation. Could potentially remove it, but not sure how important that is.

Reply:
I agree it would be a bug and if there are several files the problem the script picks the first file which does not mean it has the data expected in it so output results about active/inactive data) could be completely wrong...